drivers/timer/stm32_lptim_timer.c

```diff
@@ -403,8 +403,10 @@ static int sys_clock_driver_init(void)
 
 #if defined(LL_APB1_GRP1_PERIPH_LPTIM1)
 	LL_APB1_GRP1_ReleaseReset(LL_APB1_GRP1_PERIPH_LPTIM1);
-#elif defined(LL_APB3_GRP1_PERIPH_LPTIM1)
+#elif defined(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN)
 	LL_SRDAMR_GRP1_EnableAutonomousClock(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN);
```

Comment:
This specific change is actually a fix over the initial implementation which impacts U5. Can you make it a specific commit ?
Also, since both `LL_APB3_GRP1_PERIPH_LPTIM1` and `LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN` are defined in STM32U5, the if/elif/elif sequence would not be valid anymore on U5
I'd propose to split out `LL_SRDAMR_GRP1_EnableAutonomousClock(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN);` to a dedicated `#if defined(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN)` section, then validate that running following sequence is fine on U5:
```
LL_SRDAMR_GRP1_EnableAutonomousClock(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN);
LL_APB3_GRP1_ReleaseReset(LL_APB3_GRP1_PERIPH_LPTIM1);
```

Reply:
To be honest, I don't think the calls to `LL_APBx_GRPn_ReleaseReset()` do anything. This clears the reset bit in RCC and is usually called after setting this bit via `LL_APBx_GPRn_ForceReset()`. But I couldn't find any place that would set the reset bit and it is 0 by default. But perhaps I'm missing something?
The `LL_SRDAMR_GRP1_EnableAutonomousClock(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN);` is still useful, since it allows LPTIM operation in autonomous mode.

Reply:
It seems the `LL_APBx_GRPn_ForceReset()` was present in initial commit of this file. But was later removed in commit 0856e9915500e2138772a5553ceabc2e6d78b7d0. However the `LL_APBx_GRPn_ReleaseReset()`.
I think it would be best to remove all `LL_APBx_GRPn_ReleaseReset()` calls and keep only `LL_SRDAMR_GRP1_EnableAutonomousClock(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN);` for U5.
Should I do it as a separate PR, or separate commit inside PR is enough?

Reply:
A separate commit in this PR would be fine, thanks.